pkg/agent/sysadvisor/plugin/qosaware/qos_aware.go

```diff
@@ -105,6 +105,7 @@ func (qap *QoSAwarePlugin) Run(ctx context.Context) {
 	go wait.UntilWithContext(ctx, qap.periodicWork, qap.period)
 
 	go qap.qrmServer.Run(ctx)
+	go qap.resourceAdvisor.Run(ctx)
```

Comment:
It seems that unnecessary to create new goroutine.

Reply:
Run works in a blocking way, waiting context to be done. So a new goroutine is needed

Reply:
resourceAdvisor.Run will also create new goroutine